pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -122,7 +122,7 @@ def __init__(
         self.ipus = ipus
         self.num_nodes = num_nodes
         self.sync_batchnorm = sync_batchnorm
-        self.benchmark = benchmark
+        self.benchmark = not deterministic if benchmark is None else benchmark
```

Comment:
should this raise a misconfiguration error if someone explicitly specifies `benchmark=True, deterministic=True`?

Reply:
IMO no because if `torch` allows it, we should too.

This PR is more about "better defaults" than constraining the available options

Reply:
would this setting lead to a runtime error though? and if it does, shouldn't we either fail fast here, or indicate that `deterministic` has higher priority than `benchmark`, and override `benchmark` to be False?

Reply:
> would this setting lead to a runtime error though? 

Can you please give a case where you think a runtime error might be raised? I can't think of such a case right now, hence the question.